--- booty-0.93/bootloaderInfo.py~	2007-12-07 19:19:40.000000000 +0100
+++ booty-0.93/bootloaderInfo.py	2007-12-14 15:24:12.816148293 +0100
@@ -337,9 +337,7 @@
             initrd = booty.makeInitrd (kernelTag, instRoot)
 
             sl.addEntry("label", label)
-            if os.access (instRoot + initrd, os.R_OK):
-                sl.addEntry("initrd", "%sinitrd%s.img" %(self.kernelLocation,
-                                                         kernelTag))
+            sl.addEntry("initrd", initrd)
                 
             sl.addEntry("read-only")
 
@@ -760,7 +758,7 @@
         f.write('#          root %s\n' % self.grubbyPartitionName(bootDevs[0]))
         f.write("#          kernel %svmlinuz-version ro "
                 "root=/dev/%s\n" % (cfPath, rootDev))
-        f.write("#          initrd %sinitrd-version.img\n" % (cfPath))
+        f.write("#          initrd %sinitrd-version.gz\n" % (cfPath))
         f.write("#boot=/dev/%s\n" % (grubTarget))
 
         # get the default image to boot... we have to walk and find it
@@ -2071,12 +2071,12 @@
 # FIXME: this is based on the stuff from anaconda, but kind of crappy :-/
 def getProductName():
     # try redhat-release first
-    if os.access("/etc/redhat-release", os.R_OK):
-        f = open("/etc/redhat-release", "r")
+    if os.access("/etc/pld-release", os.R_OK):
+        f = open("/etc/pld-release", "r")
         lines = f.readlines()
         f.close()
         for buf in lines:
-            relidx = buf.find(" release")
+            relidx = buf.find("PLD Linux")
             if relidx != -1:
                 return buf[:relidx]
 
@@ -2095,6 +2095,6 @@
             return lines[1][:-1]
 
     # fall back
-    return "Red Hat Linux"
+    return "PLD Linux"
         
             
--- booty-0.93/booty.py~	2007-12-07 19:19:40.000000000 +0100
+++ booty-0.93/booty.py	2008-08-31 00:01:20.375418616 +0200
@@ -194,9 +194,6 @@
         
         initrd = makeInitrd("-%s" % (newVersion,), "/")
 
-        if not os.access(initrd, os.R_OK):
-            initrd = None
-
         if imageType and imageType != defaultType:
             # linux-smp.linux-BOOT, etc
             label = "linux-"+imageType
@@ -276,7 +273,7 @@
     if rhpl.getArch() == 'ia64':
         initrd = "/boot/efi/EFI/redhat/initrd%s.img" % (kernelTag, )
     else:
-        initrd = "/boot/initrd%s.img" % (kernelTag, )
+        initrd = "/boot/initrd%s.gz" % (kernelTag, )
 
     return initrd
 
