--- booty-0.93/bootloaderInfo.py~	2007-12-07 19:19:40.000000000 +0100
+++ booty-0.93/bootloaderInfo.py	2007-12-14 15:24:12.816148293 +0100
@@ -2071,12 +2071,12 @@
 # FIXME: this is based on the stuff from anaconda, but kind of crappy :-/
 def getProductName():
     # try redhat-release first
-    if os.access("/etc/redhat-release", os.R_OK):
-        f = open("/etc/redhat-release", "r")
+    if os.access("/etc/pld-release", os.R_OK):
+        f = open("/etc/pld-release", "r")
         lines = f.readlines()
         f.close()
         for buf in lines:
-            relidx = buf.find(" release")
+            relidx = buf.find("PLD Linux")
             if relidx != -1:
                 return buf[:relidx]
 
@@ -2095,6 +2095,6 @@
             return lines[1][:-1]
 
     # fall back
-    return "Red Hat Linux"
+    return "PLD Linux"
         
             
