--- booty-0.93/bootloaderInfo.py~	2007-12-14 15:25:30.000000000 +0100
+++ booty-0.93/bootloaderInfo.py	2007-12-14 15:27:46.301475773 +0100
@@ -710,7 +710,7 @@
 
         # XXX old config file should be read here for upgrade
 
-        cf = '/boot/grub/grub.conf'
+        cf = '/boot/grub/menu.lst'
         self.perms = 0600
         if os.access (instRoot + cf, os.R_OK):
             self.perms = os.stat(instRoot + cf)[0] & 0777
@@ -867,7 +867,7 @@
             usedDevs[device] = 1
 
         f.close()
-        os.chmod(instRoot + "/boot/grub/grub.conf", self.perms)
+        os.chmod(instRoot + "/boot/grub/menu.lst", self.perms)
 
         try:
             # make symlink for menu.lst (default config file name)
@@ -883,7 +883,7 @@
             if os.access (instRoot + "/etc/grub.conf", os.R_OK):
                 os.rename(instRoot + "/etc/grub.conf",
                           instRoot + "/etc/grub.conf.rpmsave")
-            os.symlink("../boot/grub/grub.conf", instRoot + "/etc/grub.conf")
+            os.symlink("../boot/grub/menu.lst", instRoot + "/etc/grub.conf")
         except:
             pass
        
@@ -947,7 +947,7 @@
             if target == "partition":
                 stage1Target = self.grubbyPartitionName(gtPart)
 
-            cmd += "install %s%s/stage1 d %s %s/stage2 p %s%s/grub.conf" % \
+            cmd += "install %s%s/stage1 d %s %s/stage2 p %s%s/menu.lst" % \
                 (args, grubPath, stage1Target, grubPath, bPart, grubPath)
             cmds.append(cmd)
             
@@ -1065,7 +1065,7 @@
 
         theDev = None
         for (fn, stanza) in [ ("/etc/sysconfig/grub", "boot="),
-                              ("/boot/grub/grub.conf", "#boot=") ]:
+                              ("/boot/grub/menu.lst", "#boot=") ]:
             try:
                 f = open(instRoot + fn, "r")
             except:
@@ -1138,7 +1138,7 @@
                     break
                     
             args = "--stage2=/boot/grub/stage2 "
-            cmd ="root %s\ninstall %s%s/stage1 d %s %s/stage2 p %s%s/grub.conf"\
+            cmd ="root %s\ninstall %s%s/stage1 d %s %s/stage2 p %s%s/menu.lst"\
                 % (grubbyRootPart, args, grubPath, grubbyStage1Dev,
                    grubPath, grubbyRootPart, grubPath)
         
