--- booty-0.71/butil.py	2006-06-06 01:53:52.720489440 +0300
+++ booty-0.71/butil.py	2006-06-06 02:01:46.891108363 +0300
@@ -151,15 +151,12 @@
 
 # returns a product name to use for the boot loader string
 def getProductName():
-    # try redhat-release first
-    if os.access("/etc/redhat-release", os.R_OK):
-        f = open("/etc/redhat-release", "r")
+    # try pld-release first
+    if (os.access("/etc/pld-release", os.R_OK)):
+        f = open("/etc/pld-release", "r")
         lines = f.readlines()
         f.close()
-        for buf in lines:
-            relidx = buf.find(" release")
-            if relidx != -1:
-                return buf[:relidx]
+        return lines[0]
 
     if os.access("/tmp/product/.buildstamp", os.R_OK):
         path = "/tmp/product/.buildstamp"
@@ -177,5 +174,5 @@
 
     # fall back
-    return "Red Hat Linux"
+    return "PLD Linux"
         
             
